src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py

```diff
@@ -240,6 +259,11 @@ def _upload_and_run_pipeline(
                 logger.info(
                     "Started one-off pipeline run with ID '%s'.", result.run_id
                 )
+
+                if self.synchronous:
```

Comment:
I just thought I'd mention this: this is something I often needed even as an end user: forcing zenml to wait locally until the remote pipeline execution is complete. I understand this is only meant for integration testing, but it would be nice to have it exposed in the future as a feature that users can also access and use, maybe as a pipeline run config attribute.

Reply:
As you said it's not really intended for users right now (e.g. fixed timeout), but if you see a need for that it would certainly be an easy change. I'll add a todo for now if that's okay with you

Reply:
a TODO was exactly what I was looking for